### ISSUE TITLE
autoscale instance count from 1 to 2 only for purposes of service update

### ISSUE THREAD

**Issue by ssozonoff** [silinternational/ecs-deploy#120]
Do you know if its possible to have the instance count scale say from 1 to 2 when we perform a service update and then scale back down to 1 once the update is complete ?

**Comment by fillup** [comment]
I like that idea, and I would assume its possible, although the solution may be fairly specific to your situation. For example, to keep the math simple, lets say you're running on a single ec2 instance and using almost all the memory/cpu. In your auto scaling group you could set it that when cpu or memory reservation got to >= 100%, scale up by one instance. Then when the cpu / memory reservation got back to <= 50% (of new two instance capacity), scale down by one instance. I think this would cause the desire to place a new task launch another instance, and once that instance was running that the task was running and the old task was stopped, it would want to scale down again. This is an over simplified example and not proven, but think it through and see if maybe it could work for your situation.
